plugins/woocommerce/src/Internal/Admin/ProductReviews/ReviewsUtil.php

```diff
@@ -10,17 +10,52 @@ class ReviewsUtil {
 	/**
 	 * Removes product reviews from the edit-comments page to fix the "Mine" tab counter.
 	 *
-	 * @param  array|mixed $clauses A compacted array of comment query clauses.
+	 * @param array|mixed       $clauses A compacted array of comment query clauses.
+	 * @param \WP_Comment_Query $comment_query The WP_Comment_Query instance being filtered.
+	 *
 	 * @return array|mixed
 	 */
-	public static function comments_clauses_without_product_reviews( $clauses ) {
-		global $wpdb, $current_screen;
+	public static function comments_clauses_without_product_reviews( $clauses, $comment_query ) {
+		global $wpdb;
+
+		if ( ! empty( $comment_query->query_vars['post_type'] ) ) {
+			$post_type = $comment_query->query_vars['post_type'];
+			if ( ! is_array( $post_type ) ) {
+				$post_type = explode( ',', $post_type );
+			}
+			if ( ! in_array( 'product', $post_type ) ) {
+				return $clauses;
+			}
+		}
+
+		/**
+		 * Any comment queries with these values are likely to be custom handling where we don't want to change default behavior.
+		 */
+		foreach ( [ 'ID', 'parent', 'parent__in', 'post_author__in', 'post_author', 'post_name', 'type', 'type__in', 'type__not_in', 'post_type__in' ] as $arg ) {
```

Comment:
Should we also add `comment__in` and `comment__not_in` to this list of arguments?